src/Compilers/CSharp/Test/Symbol/Symbols/StaticAbstractMembersInInterfacesTests.cs

```diff
@@ -0,0 +1,4145 @@
+                //     sealed static bool P01 { get => false; }
+                Diagnostic(ErrorCode.ERR_SealedNonOverride, "P01").WithArguments("C1.P01").WithLocation(5, 24),
+                // (6,39): error CS0238: 'C1.E01' cannot be sealed because it is not an override
+                //     sealed static event System.Action E01 { add {} remove {} }
+                Diagnostic(ErrorCode.ERR_SealedNonOverride, "E01").WithArguments("C1.E01").WithLocation(6, 39),
+                // (7,37): error CS0106: The modifier 'sealed' is not valid for this item
+                //     public sealed static C1 operator+ (C1 x) => null;
+                Diagnostic(ErrorCode.ERR_BadMemberFlag, "+").WithArguments("sealed").WithLocation(7, 37)
+                );
+        }
+
+        [Fact]
+        public void AbstractStaticInStruct_01()
+        {
+            var source1 =
+@"
+struct C1
+{
+    public abstract static void M01();
+    public abstract static bool P01 { get; }
+    public abstract static event System.Action E01;
+    public abstract static C1 operator+ (C1 x);
+}
+";
+            var compilation1 = CreateCompilation(source1, options: TestOptions.DebugDll,
+                                                 parseOptions: TestOptions.RegularPreview,
+                                                 targetFramework: TargetFramework.NetCoreApp);
+
+            compilation1.VerifyDiagnostics(
+                // (4,33): error CS0112: A static member cannot be marked as 'abstract'
+                //     public abstract static void M01();
+                Diagnostic(ErrorCode.ERR_StaticNotVirtual, "M01").WithArguments("abstract").WithLocation(4, 33),
+                // (5,33): error CS0112: A static member cannot be marked as 'abstract'
+                //     public abstract static bool P01 { get; }
+                Diagnostic(ErrorCode.ERR_StaticNotVirtual, "P01").WithArguments("abstract").WithLocation(5, 33),
+                // (6,48): error CS0112: A static member cannot be marked as 'abstract'
+                //     public abstract static event System.Action E01;
+                Diagnostic(ErrorCode.ERR_StaticNotVirtual, "E01").WithArguments("abstract").WithLocation(6, 48),
+                // (7,39): error CS0106: The modifier 'abstract' is not valid for this item
+                //     public abstract static C1 operator+ (C1 x);
+                Diagnostic(ErrorCode.ERR_BadMemberFlag, "+").WithArguments("abstract").WithLocation(7, 39),
+                // (7,39): error CS0501: 'C1.operator +(C1)' must declare a body because it is not marked abstract, extern, or partial
+                //     public abstract static C1 operator+ (C1 x);
+                Diagnostic(ErrorCode.ERR_ConcreteMissingBody, "+").WithArguments("C1.operator +(C1)").WithLocation(7, 39)
+                );
+        }
+
+        [Fact]
+        public void SealedStaticInstruct_01()
+        {
+            var source1 =
+@"
+class C1
```

Comment:
Was this meant to be a struct?

Reply:
> Was this meant to be a struct?

Good catch, I'll adjust this in th enext PR.

---
In reply to: [606004921](https://github.com/dotnet/roslyn/pull/52228#discussion_r606004921) [](ancestors = 606004921)